File: runme/mcgen/local/datan/exasrc/e7mtx.c
#include <stdio.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__1 = 1, c__3 = 3;
static double c_d0 = 0.;

main(void)
{
    /* Initialized data */
    static logical ok;
    static double a[9] /* was [3][3] */ = { 1.,2.,1.,2.,1.,1.,3.,-2.,2. },
                  b[9] /* was [3][3] */ = { 1.,0.,0.,0.,1.,0.,0.,0.,1. },
                  r[3], x[9] /* was [3][3] */;
    static integer nn;

/* demonstrates solution of matrix equation */
/* by singular value decomposition (SVD) */
/* identify program to user */
    printf("%s\n", "Program E7MTX demonstrates use of");
    printf("%s\n\n", "MTXSVD for singular value decomposition.");
    printf("%s\n", "Output produced by program E7MTX.");
    printf("%s\n", "------------------------------------------");
/* write input data */
    printf("%s%2li%s%2li\n", "N = ", c__3, ", M = ", c__3);
/* write initial matrices */
    printf("%s\n", "A =");
    mtxwrt_(a, &c__3, &c__3);
    printf("%s\n", "B =");
    mtxwrt_(b, &c__3, &c__3);
    printf("%s\n\n", "------------------------------------------");
/* demonstrate MTXSVD */
    nn = c__3;
    mtxsvd_(a, b, x, r, &c__3, &nn, &c__3, &c_d0, &ok);
    printf("%s\n", "CALL MTXSVD(A,B,X,R,N,N,M,FRAC,OK) yields");
    printf("%s%li\n", "OK =", ok);
    printf("%s\n", "X =");
    mtxwrt_(x, &c__3, &c__3);
    printf("%s\n", "R =");
    mtxwrt_(r, &c__1, &c__3);
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e7mtx.f
      PROGRAM E7MTX
C demonstrates solution of matrix equation
C by singular value decomposition (SVD)
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      PARAMETER (N=3,M=3,FRAC=0.D0)
      LOGICAL OK
      DIMENSION A(N,N),B(N,M),X(N,M),R(N)
      DATA A/1.D0,2.D0,1.D0,2.D0,1.D0,1.D0,3.D0,-2.D0,2.D0/
      DATA B/1.D0,0.D0,0.D0,0.D0,1.D0,0.D0,0.D0,0.D0,1.D0/
C identify program to user
      WRITE(*,*)' Program E7MTX demonstrates use of'
      WRITE(*,*)' MTXSVD for singular value decomposition'
      WRITE(*,*)' '
      WRITE(*,*)' Output produced by program E7MTX'
      WRITE(*,'(A)')' ------------------------------------------'
C write input data
      WRITE(*,'(A,I2,A,I2)')' N = ',N,', M = ',M
C write initial matrices
      WRITE(*,*)' A ='
      CALL MTXWRT(A,N,N)
      WRITE(*,*)' B ='
      CALL MTXWRT(B,N,M)
      WRITE(*,'(A)')' ------------------------------------------'
      WRITE(*,*)' '
C demonstrate MTXSVD
      NN=N
      CALL MTXSVD(A,B,X,R,N,NN,M,FRAC,OK)
      WRITE(*,'(A)')' CALL MTXSVD(A,B,X,R,N,N,M,FRAC,OK) yields '
      WRITE(*,*)' OK =',OK
      WRITE(*,*)' X ='
      CALL MTXWRT(X,N,N)
      WRITE(*,*)' R ='
      CALL MTXWRT(R,1,N)
      END
